pkg/diagnostic/webhook_server/check_test.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: Copyright (c) 2016-2024, CloudZero, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package webhook_server_test
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/cloudzero/cloudzero-agent-validator/pkg/config"
+	"github.com/cloudzero/cloudzero-agent-validator/pkg/diagnostic/webhook_server"
+	"github.com/cloudzero/cloudzero-agent-validator/pkg/status"
+	"github.com/stretchr/testify/assert"
+)
+
+func makeReport() status.Accessor {
+	return status.NewAccessor(&status.ClusterStatus{})
+}
+
+func TestCheck(t *testing.T) {
```

Comment:
I didn't use the `MockRoundTripper` setup that the other tests use because I needed to change the `TLSClientConfig` in the client.Transport. I wasn't sure of a nice way to do that while keeping the method signature consistent.